libraries/logrotate.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Cookbook:: logrotate
+# Library:: logrotate
+#
+# Copyright:: 2013-2019, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Logrotate
+  module Cookbook
+    module LogrotateHelpers
+      OPTIONS ||= %w(compress copy copytruncate daily dateext
+                     dateyesterday delaycompress hourly ifempty mailfirst maillast
+                     missingok monthly nocompress nocopy nocopytruncate nocreate nocreateolddir
+                     nodelaycompress nodateext nomail nomissingok noolddir
+                     nosharedscripts noshred notifempty renamecopy sharedscripts shred weekly
+                     yearly).freeze
+
+      PARAMETERS ||= %w(compresscmd uncompresscmd compressext compressoptions
+                        create createolddir dateformat include mail extension maxage minsize maxsize
+                        rotate size shredcycles start tabooext su olddir).freeze
+
+      SCRIPTS ||= %w(firstaction prerotate postrotate lastaction preremove).freeze
+
+      PARAMETERS_AND_OPTIONS_AND_SCRIPTS ||= PARAMETERS + OPTIONS + SCRIPTS
+
+      def parameters_from(hash)
+        hash.select { |k, _| PARAMETERS.include?(k) }
+      end
+
+      def options_from(values)
+        values.sort.select { |v| OPTIONS.include?(v) }
+      end
+
+      def paths_from(hash)
+        hash.select { |k| !PARAMETERS_AND_OPTIONS_AND_SCRIPTS.include?(k) }.map do |path, config|
```

Comment:
You might be better off doing a do, end block here for readability

Reply:
```ruby
      def paths_from(hash)
        hash.select do |k|
          !PARAMETERS_AND_OPTIONS_AND_SCRIPTS.include?(k)
        end.map do |path, config|
          [path, { 'parameters' => parameters_from(config), 'options' => options_from(config), 'scripts' => scripts_from(config) }]
        end.to_h
      end
```

I can't decide if that's any better or not, I am a big fan of the ruby one line block though so it might be me.

Reply:
Infact it hits this rubocop (https://rubystyle.guide#single-line-blocks) so I think that one stays as it's not me.

Reply:
Definetly a prime candidate for core and it'd probably be worth re-implementing it anyway as the resource/provider format is so different.